modules/system/classes/extensions/PluginManager.php

```diff
@@ -1034,41 +1421,83 @@ protected function sortByDependencies(): array
         return $this->plugins = $sortedPlugins;
     }
 
-    //
-    // Management
-    //
+    public function resolveIdentifier(ExtensionSource|WinterExtension|string $extension): ?string
```

Comment:
Should we be more specific in the manager resolveIdentifier() methods to only accept the specific WinterExtension implementations that we accept?

Reply:
It's overriding the parent, not sure if we can limit the scope of the type hints